js/packages/teams-ai/src/planners/ActionPlanner.ts

```diff
@@ -262,10 +262,27 @@ export class ActionPlanner<TState extends TurnState = TurnState> implements Plan
             startStreamingMessage: this._options.startStreamingMessage
         });
 
+        this.addActionOutputs(memory, history_variable, client);
+
         // Complete prompt
         return await client.completePrompt(context, memory, this.prompts);
     }
 
+    private addActionOutputs(memory: Memory, historyVariable: string, client: LLMClient): void {
+        const history: Message[] = memory.getValue(historyVariable);
+
+        if (history && history.length > 1) {
+            // Submit action outputs
+            const actionOutputs: string[] = memory.getValue('temp.actionOutputs') ?? [];
+            const actionCalls: ActionCall[] = history[history.length - 1].action_calls ?? [];
+
+            for (const actionCall of actionCalls) {
+                const output = actionOutputs[actionCall.id as unknown as number];
```

Comment:
note to check for `string` instead

Reply:
This is actually necessary as number  because the number if checking for the array.